.github/workflows/release.yml

```diff
@@ -19,9 +19,12 @@ on:
           # - premajor
   pull_request:
     types: [closed]
+    branches:
+      - main
 
 jobs:
   release:
+    if: github.event.pull_request.merged == true
```

Comment:
Docs for this: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-your-pull_request-workflow-when-a-pull-request-merges